activerecord/lib/active_record/connection_adapters/postgresql/quoting.rb

```diff
@@ -133,6 +137,10 @@ def encode_array(array_data)
             result
           end
 
+          def encode_range(range)
+            "[#{quote(range.first)},#{quote(range.last)}#{range.exclude_end? ? ')' : ']'}"
```

Comment:
I don't think we need to call `quote` here

Reply:
Calling `quote` makes sure the values are properly formatted for queries. For example: removing `quote`, causes the microseconds to be dropped

```ruby
(byebug) from = type_cast_single_for_database(range.first)
2010-01-01 13:30:00 UTC

(byebug) to = type_cast_single_for_database(range.last)
2011-02-02 19:30:00 UTC

(byebug) "[#{from},#{to}#{value.exclude_end? ? ')' : ']'}"
"[2010-01-01 13:30:00 UTC,2011-02-02 19:30:00 UTC)"

(byebug) "[#{quote(from)},#{quote(to)}#{value.exclude_end? ? ')' : ']'}"
"['2010-01-01 13:30:00.670277','2011-02-02 19:30:00.745125')"
```

Let me know if there's another place where we should call `quote` on the values, but we need to call `quote` in order to make sure the values are properly formatted and nothing in lost. I think calling it in `encode_range` makes the most sense, because it's part of building the range.

Reply:
Calling `type_cast` should do the same thing